cli/cmd/plugin/ui/web/tanzu-ui/src/views/unmanaged-cluster/UnmanagedClusterInventory.test.tsx

```diff
@@ -1,17 +1,142 @@
+                ctx.status(200),
+                ctx.json([
+                    {
+                        name: 'work-space-cluster',
+                        provider: 'MiniKube',
+                        status: 'Running',
+                    },
+                    {
+                        name: 'tanzu-cluster',
+                        provider: 'Kind',
+                        status: 'Stopped',
+                    },
+                    {
+                        name: 'ui-cluster',
+                        provider: 'Kind',
+                        status: 'Unknown',
+                    },
+                ])
+            );
+        })
+    );
+
+    beforeAll(() => server.listen({ onUnhandledRequest: 'bypass' }));
+    afterEach(() => server.resetHandlers());
+    afterAll(() => server.close());
+
+    const clickFirstDeleteButton = async function () {
+        const deleteBtns = await screen.findAllByText('Delete');
+        const firstDeleteBtn = deleteBtns[0];
+        fireEvent.click(firstDeleteBtn);
+    };
+
+    test('should render', async () => {
+        const view = render(<UnmanagedClusterInventory />);
+        await waitFor(() => {
+            expect(view).toBeDefined();
+        });
+    });
+
+    test('should always display a button to create an unmanaged cluster', async () => {
+        render(<UnmanagedClusterInventory />);
+
+        expect(await screen.findByText('create unmanaged cluster')).toBeInTheDocument();
+    });
+
+    test('should display three unmanaged cluster cards when unmanaged clusters are present', async () => {
+        render(<UnmanagedClusterInventory />);
+        const unmanagedClusterCards = await screen.findAllByTestId('unmanaged-cluster-card');
+        expect(unmanagedClusterCards.length).toBe(3);
+    });
+
+    test('should display name, provider and status for an unmanaged cluster card', async () => {
+        render(<UnmanagedClusterInventory />);
+        expect(await screen.findByText('work-space-cluster')).toBeInTheDocument();
+        expect(await screen.findByText('MiniKube')).toBeInTheDocument();
+        expect(await screen.findByText('Running')).toBeInTheDocument();
+    });
+
+    test('delete button should open modal confirmation', async () => {
+        render(<UnmanagedClusterInventory />);
+
+        await clickFirstDeleteButton();
+
+        expect(await screen.findByTestId('confirm-delete-cluster-modal')).toBeInTheDocument();
+    });
+
+    test('delete modal confirmation cancel button should close modal window', async () => {
+        render(<UnmanagedClusterInventory />);
+
+        await clickFirstDeleteButton();
+
+        const cancelBtn = await screen.findByText('Cancel');
+        fireEvent.click(cancelBtn);
+        expect(screen.queryByTestId('confirm-delete-cluster-modal')).not.toBeInTheDocument();
+
+        const umanagedClusterCards = await screen.findAllByTestId('unmanaged-cluster-card');
+        expect(umanagedClusterCards.length).toBe(3);
```

Comment:
excellent to test that the number of cards has not changed! However, see note above about using the size of the test data array